amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -193,20 +195,26 @@ public void run() {
             return;
         }
 
+        Exception exceptionToThrow = null;
         try {
             initialize();
             log.info("Initialization complete. Starting worker loop.");
         } catch (RuntimeException e) {
             log.error("Unable to initialize after {} attempts. Shutting down.", lifecycleConfig.maxInitializationAttempts(), e);
             shutdown();
+            exceptionToThrow = e;
         }
 
         while (!shouldShutdown()) {
+            exceptionToThrow = null;
```

Comment:
Need not set this. If exception is caught from initialize, shutdown is invoked and shouldShutdown will return true and the code should not enter this loop.

Reply:
Will delete